.github/workflows/tpu_ci.yml

```diff
@@ -29,8 +29,12 @@ jobs:
       - name: Run Tests
         env:
           PJRT_DEVICE: TPU
+        # TODO: figure out how to update the JAX version automatically when the libtpu version is updated.
+        # Jax is needed for pallas tests.
         run: |
           pip install fsspec
           pip install rich
+          pip install -U --pre jaxlib==0.4.25.dev20240213 -f https://storage.googleapis.com/jax-releases/jaxlib_nightly_releases.html
```

Comment:
Can you add these dependencies to our `setup.py`? Maybe a `pallas` `extras_require` since AFAIK `jax` will still be optional. We can update the JAX version together with `libtpu-nightly` when we do pin updates

Reply:
Sure, I wasn't quite sure to make it default for development, but I guess why not.

Reply:
@will-cromar How are you going to add nightly dependencies to setup.py? I tried:
```
ptxla@t1v-n-94bbce5b-w-0:/workspaces/work/pytorch/xla$ git diff setup.py 
diff --git a/setup.py b/setup.py
index e4fdaebee..66a1eb321 100644
--- a/setup.py
+++ b/setup.py
@@ -288,6 +288,8 @@ setup(
         # importlib.metadata backport required for PJRT plugin discovery prior
         # to Python 3.10
         'importlib_metadata>=4.6;python_version<"3.10"',
+        'jaxlib==0.4.25.dev20240213 @ https://storage.googleapis.com/jax-releases/jaxl
ib_nightly_releases.html',
+        'jax==0.4.25.dev20240213 @ https://storage.googleapis.com/jax-releases/jax_nig
htly_releases.html',
     ],
     package_data={
         'torch_xla': ['lib/*.so*',],
```

but that doesn't work...
```
error in torch_xla setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at "'@ https:'": Expected string_end
```

Reply:
Add it to [`extras_require`](https://github.com/pytorch/xla/blob/6cc5c3819c09c7b1b4ca4927d7fa65133f95b41c/setup.py#L301) so it's not required by default:

```
    extras_require={
        # On Cloud TPU VM install with:
        # pip install torch_xla[tpu] -f https://storage.googleapis.com/libtpu-releases/index.html
        'tpu': [f'libtpu-nightly=={_libtpu_version}'],
        # On nightly, install libtpu with `pip install torch_xla[tpuvm]`
        # Remove from release branches since this is not allowed by PyPI.
        'tpuvm': [f'libtpu-nightly @ {_libtpu_storage_path}'],
        'pallas': ['jaxlib==0.4.25.dev20240213', 'jax==0.4.25.dev20240213']
    },
```

To uploadable to PyPI, you can't bake in a link to another index. The user will have to do `pip install torch_xla[pallas] -f path/to/index` like they do for TPU.

.... unless you can just rely on a stable release of `jax` that's already on PyPI. If JAX is not loading libtpu (I suspect it does not, otherwise we would crash), the JAX build date does not have to match `libtpu-nightly`. Maybe better to investigate before we make the feature stable.

Reply:
No, JAX will need Mosaic which is in the libtpu to generate the payload. So we need the nightly. Thanks, Will. Will do it in this way.